### PR TITLE
dev: upgrade to react 18.3.x and wp-6.6 across all projects

### PR DESCRIPTION
**Pull request by rjchow**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR had the aim of addressing https://github.com/woocommerce/woocommerce/issues/52189

However, it was noted that bumping the package.json React dependency to 18 required a few additional supporting changes:
- @wordpress/* dependencies had to be upgraded to versions that use React 18 as well, otherwise there would be multiple versions of React and causing build failures. I picked WP 6.6 rather than the minimum version of WP (6.2) that had React 18, to save us the trouble of migrating yet again later.
- @testing-library/* had to be updated as well, since the existing version used React 17
- TS Checking was disabled in several packages, for which I will create issues to fix the type errors and re-enable TS
- TS errors were converted to warnings for @woocommerce/admin-library, for which we will need a few issues to delegate to various teams
- Resulting from the above changes, there were test failures and type errors that had to be fixed in order to get a green CI. I have fixed a large part of it where I could, but there are some remaining ones that I could not fix without spending time getting familiar with the expected behaviour of the code. These I tested in the frontend to ensure there was no breakage, and tagged them with `// TODO: react-18-upgrade - <reason>`. They will also be delegated.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. I'm not really sure there are specific testing instructions, but a comprehensive smoke test would be prudent.
2. It's not expected that there are changes in the built files for WC Core, and Admin. It seems like there might be changes in Blocks, but this is pending @woocommerce/rubik's closer look
3. Build should complete successfully, and there should be a couple hundred TS warnings for Admin.

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [x] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [x] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

Updated declared dependencies to React 18 and Wordpress 6.6

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: .syncpackrc [hunk 1]
```diff
@@ -119,14 +119,33 @@
 			"dependencies": [
 				"react",
 				"react-dom",
+				"react-test-renderer",
 				"@types/react-dom",
 				"@types/react"
 			],
-			"pinVersion": "17.0.x",
+			"pinVersion": "18.3.x",
```
Comment (chihsuan) [inline .syncpackrc:126]: There is a `React-related dependencies: React 18` above. Is it possible to merge them into one?

File: package.json [hunk 2]
```diff
@@ -65,18 +65,17 @@
...
 	"pnpm": {
 		"overrides": {
-			"@types/react": "17.0.x",
-			"@wordpress/data": "wp-6.0",
-			"@wordpress/rich-text": "wp-6.0",
-			"react-resize-aware": "3.1.1",
-			"@automattic/tour-kit>@wordpress/element": "4.4.1"
+			"@types/react": "18.3.x",
+			"@wordpress/data": "wp-6.6",
+			"@wordpress/rich-text": "wp-6.6",
```
Comment (chihsuan) [inline package.json]: I added `@wordpress/rich-text` to the overrides because higher version of `@wordpress/rich-text` is not compatible with `@wordpress/data@wp-6.0`. I think we can remove it now since we are using `@wordpress/data@wp-6.6` (Need to test it).

File: packages/js/block-templates/src/hooks/use-layout-template/use-layout-template.ts [hunk 3]
```diff
@@ -2,17 +2,7 @@
...
-// @ts-ignore No types for this exist yet, natively (not until 7.0.0).
-// Including `@types/wordpress__data` as a devDependency causes build issues,
-// so just going type-free for now.
-// eslint-disable-next-line @woocommerce/dependency-group
-import { useEntityRecord } from '@wordpress/core-data';
-// eslint-disable-next-line @typescript-eslint/ban-ts-comment
-// @ts-ignore No types for this exist yet, natively (not until 7.0.0).
-// Including `@types/wordpress__data` as a devDependency causes build issues,
-// so just going type-free for now.
-// eslint-disable-next-line @woocommerce/dependency-group
```
Comment (chihsuan) [inline packages/js/block-templates/src/hooks/use-layout-template/use-layout-template.ts:15]: Nice! 😍

File: packages/js/data/src/countries/index.ts [hunk 4]
```diff
@@ -36,7 +36,7 @@ declare module '@wordpress/data' {
 	function select(
 		key: typeof STORE_NAME
 	): SelectFromMap< typeof selectors > & WPDataSelectors;
-	function resolveSelect(
-		key: typeof STORE_NAME
-	): PromiseifySelectors< SelectFromMap< typeof selectors > >;
+	// function resolveSelect(
```
Comment (chihsuan) [inline packages/js/data/src/countries/index.ts]: Should we add a `@ts-expect-error -- TODO: react-18-upgrade` for this? Or can we remove this?

File: packages/js/notices/src/store/index.ts [hunk 5]
```diff
@@ -17,6 +17,7 @@ export * from './types';
 export const STORE_NAME = 'core/notices2';
 // NOTE: This uses core/notices2, if this file is copied back upstream
 // to Gutenberg this needs to be changed back to core/notices.
+// @ts-expect-error - registerStore is not typed
```
Comment (chihsuan) [inline packages/js/notices/src/store/index.ts]: Ah, looks like registerStore is deprecated. We should use `register` instead. We can create a follow-up issue to migrate it.

https://developer.wordpress.org/block-editor/reference-guides/packages/packages-data/#registerstore

File: packages/js/product-editor/src/components/more-menu-dropdown/index.tsx [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+import classnames from 'classnames';
+import { DropdownMenu } from '@wordpress/components';
+import { createElement } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { moreVertical } from '@wordpress/icons';
+
+/**
+ * Copied exactly from https://github.com/WordPress/gutenberg/blob/e89fceffab765902e0162a1e3359d88501e72005/packages/interface/src/components/more-menu-dropdown/index.js
+ * to unblock progress as it was discovered to have been deleted while doing a monorepo WordPress dependencies upgrade.
+ * It was deleted in https://github.com/WordPress/gutenberg/pull/59096
```
Comment (chihsuan) [inline packages/js/product-editor/src/components/more-menu-dropdown/index.tsx:13]: 👍

File: plugins/woocommerce/client/admin/client/task-lists/fills/components/load-sample-product-modal.tsx [hunk 7]
```diff
@@ -2,8 +2,7 @@
  * External dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { Modal } from '@wordpress/components';
-import { Spinner } from '@wordpress/components/build/ui';
+import { Modal, Spinner } from '@wordpress/components';
```
Comment (chihsuan) [inline plugins/woocommerce/client/admin/client/task-lists/fills/components/load-sample-product-modal.tsx:5]: It's great to have use same Spinner across the codebase, but just wanted to point out that the `Spinner` from `@wordpress/components/build/ui` is not the same as the `Spinner` from `@wordpress/components`. I used the UI Spinner to match the design in https://github.com/woocommerce/woocommerce/pull/32815.

After the change, you can see that the Spinner looks different.

https://github.com/user-attachments/assets/42d7424f-91fb-439a-b925-4a46b836c189

**Review by chihsuan — COMMENTED** [review]
Excellent work! Excited to see the upgrade. 🎉 

Just left some comments for changes first. I'll perform the smoke tests and see if there are any issues.

**Comment by chihsuan** [comment]
> - TS Checking was disabled in several packages, for which I will create issues to fix the type errors and re-enable TS
> - TS errors were converted to warnings for @woocommerce/admin-library, for which we will need a few issues to delegate to various teams

I see that we're still using `@types/wordpress*`. I think we can remove them when we start addressing the type errors since `@wordpress/*` should have the types now. I guess we might get more type errors when we remove them, though. 😅 

Related:
- https://github.com/woocommerce/woocommerce/issues/37913

**Review by chihsuan — COMMENTED** [review]
I found a bug https://github.com/woocommerce/woocommerce/issues/53743 related to CYS, but it's not associated with this PR, as I can reproduce it on the trunk.

Other than that, I tested the Core Profiler, task list, Analytics, Settings, and Marketing pages and didn't find any issues.

File: .syncpackrc (hunk 1 above)
Comment (rjchow) [reply .syncpackrc:126]: Ah yea I was meaning to get back to this - I think I'll leave them separate but rename them to "@woocommerce/blocks-library related React versions" and "Default React versions". How does that sound?

File: packages/js/block-templates/src/hooks/use-layout-template/use-layout-template.ts (hunk 3 above)
Comment (rjchow) [reply packages/js/block-templates/src/hooks/use-layout-template/use-layout-template.ts:15]: You'll be pleased to know there are more than a 100 more instances of these to remove 😆

File: packages/js/data/src/countries/index.ts (hunk 4 above)
Comment (rjchow) [reply packages/js/data/src/countries/index.ts]: Thanks! I missed this

File: packages/js/components/src/experimental-select-tree-control/test/select-tree.test.tsx [hunk 8]
```diff
@@ -76,7 +76,7 @@ describe( 'SelectTree', () => {
 			<SelectTree { ...DEFAULT_PROPS } />
 		);
 		expect( queryByText( 'Item 1' ) ).not.toBeInTheDocument();
-		queryByRole( 'combobox' )?.focus();
+		userEvent.click( queryByRole( 'combobox' )! );
```
Comment (ilyasfoo) [inline packages/js/components/src/experimental-select-tree-control/test/select-tree.test.tsx]: Was the test failing? It's a small change in behaviour too from focus to click. The test title here is `should show the popover only when focused`, might be somewhat misleading

File: packages/js/internal-js-tests/src/setup-react-testing-library.js [hunk 9]
```diff
@@ -3,3 +3,21 @@
  */
 import '@wordpress/jest-console';
 import '@testing-library/jest-dom';
+
+/**
+ * Ignore messages that match the pattern `Store ".*" is already registered.`
+ * It's expected to have duplicate registration in the test environment as there are multiple instances of the same store such as core-data, core/block-editor, etc.
+ */
+const consoleErrorSpy = jest
```
Comment (ilyasfoo) [inline packages/js/internal-js-tests/src/setup-react-testing-library.js:11]: Do you think this will eventually be resolved, or would this potentially always be required?

**Review by ilyasfoo — COMMENTED** [review]
Leaving partial review today, so far smoke tests passing on wc-admin screens, apart from the CYS issue identified by @chihsuan

File: packages/js/internal-js-tests/src/setup-react-testing-library.js (hunk 9 above)
Comment (rjchow) [reply packages/js/internal-js-tests/src/setup-react-testing-library.js:11]: I'm not sure tbh, it doesn't sound like it's something that will be resolved - I copied it from Chi's [usage in woocommerce-blocks](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-blocks/tests/js/setup-after-env.ts#L8).

On the other hand I did notice one of the test suites I was trying to fix had the issue unexpectedly resolved when I rewrote the tests completely so [that each test registered its own registry](https://github.com/woocommerce/woocommerce/pull/53531/files#diff-12d0a8b1b5896044d3f37346cb43f74bc3f4213b6a3f56f57a198a5cbff0a779R39). So that may be the solution, since I couldn't find gutenberg or calypso doing this.

It could be worth a follow up look into!

File: packages/js/components/src/experimental-select-tree-control/test/select-tree.test.tsx (hunk 8 above)
Comment (rjchow) [reply packages/js/components/src/experimental-select-tree-control/test/select-tree.test.tsx]: I changed it back to focus and added act() around the events, thanks for catching this!

**Comment by ralucaStan** [comment]
Yesterday `pnpm-lock.yaml` had changes after I tested the branch. Today, after I updated the branch and ran `pnpm clean` 3 files were deleted, see below, and the lock file still has a change:
- `woocommerce/plugins/woocommerce/src/Vendor/.gitignore`
- `woocommerce/plugins/woocommerce/src/Vendor/README.md`
- `woocommerce/plugins/woocommerce/tests/legacy/framework/vendor/class-wp-test-spy-rest-server.php`

**Comment by rjchow** [comment]
> Yesterday `pnpm-lock.yaml` had changes after I tested the branch. Today, after I updated the branch and ran `pnpm clean` 3 files were deleted, see below, and the lock file still has a change:
> 
>     * `woocommerce/plugins/woocommerce/src/Vendor/.gitignore`
> 
>     * `woocommerce/plugins/woocommerce/src/Vendor/README.md`
> 
>     * `woocommerce/plugins/woocommerce/tests/legacy/framework/vendor/class-wp-test-spy-rest-server.php`

Hey @ralucaStan thanks for reviewing! 

I did notice this - but I managed to replicate it both with various versions of pnpm and also on trunk (9.3.0, 9.13, 9.14, 9.15, etc) so I don't think its a result of this PR. Do think it's suboptimal and should be looked into, though.

**Comment by ralucaStan** [comment]
> Do think it's suboptimal and should be looked into, though.

this should be highlighted to the monorepo team

File: plugins/woocommerce/client/admin/webpack.config.js [hunk 10]
```diff
@@ -182,7 +182,31 @@ const webpackConfig = {
...
+				'TS2305', // ~4: Module has no exported member.
+				'TS2347', // ~3: Untyped function calls may not accept type arguments.
+				'TS2769', // ~2: No overload matches this call.
+				'TS2558', // ~2: Unexpected number of type arguments.
+				'TS2344', // ~2: Type does not satisfy a specific constraint.
+				'TS18046', // ~2: Variable is of type 'unknown'.
+				'TS2677', // 1: Type predicate's type must be assignable to its parameter's type.
+				'TS2571', // 1: Object is of type 'unknown'.
+				'TS2352', // 1: Invalid type conversion.
+				'TS18049', // 1: Nullable or undefined value encountered.
```
Comment (ralucaStan) [inline plugins/woocommerce/client/admin/webpack.config.js:209]: To confirm, these errors are the ones that appeared as a result of PR?

Comment (ralucaStan) [reply plugins/woocommerce/client/admin/webpack.config.js:209]: I don't see TS2352, TS2604 & TS2677 errors when running `npx tsc --noEmit` on this branch

File: packages/js/notices/src/store/index.ts (hunk 5 above)
Comment (rjchow) [reply packages/js/notices/src/store/index.ts]: some good references here: https://github.com/Automattic/wp-calypso/issues/74399

File: plugins/woocommerce/client/admin/webpack.config.js (hunk 10 above)
Comment (rjchow) [reply plugins/woocommerce/client/admin/webpack.config.js:209]: > To confirm, these errors are the ones that appeared as a result of PR?

Yep, since running `npx tsc` on trunk doesn't report anything

> I don't see TS2352, TS2604 & TS2677 errors when running npx tsc --noEmit on this branch

It's varied a little since I run those numbers about a week+ ago with changes on top of that both from the PR and trunk. Here's the latest output I have:

command: `npx tsc | awk '/error TS[0-9]+:/ {sub(/^.*error /, ""); split($0, arr, ": "); count[arr[1]]++; desc[arr[1]] = arr[2]} END {for (e in count) printf "%d %s: %s\n", count[e], e, desc[e]}' | sort -nr`

```
107 TS2349: This expression is not callable.
70 TS2578: Unused '@ts-expect-error' directive.
62 TS2554: Expected 2 arguments, but got 1.
58 TS2339: Property 'getActivePlugins' does not exist on type 'never'.
41 TS7006: Parameter 'fills' implicitly has an 'any' type.
31 TS2322: Type '{ id
25 TS2503: Cannot find namespace 'Slot'.
11 TS2559: Type '{ children
10 TS2786: 'ConnectForm' cannot be used as a JSX component.
10 TS2604: JSX element type 'ConnectForm' does not have any construct or call signatures.
7 TS2345: Argument of type 'number | undefined' is not assignable to parameter of type 'string'.
6 TS2741: Property 'label' is missing in type '{ checked
4 TS2347: Untyped function calls may not accept type arguments.
4 TS2305: Module '"@wordpress/notices"' has no exported member 'Options'.
2 TS2769: No overload matches this call.
2 TS2558: Expected 0 type arguments, but got 1.
2 TS2344: Type 'Product | ProductVariation' does not satisfy the constraint 'StoreDescriptor<any> | MapSelect'.
2 TS18046: 'a' is of type 'unknown'.
1 TS2677: A type predicate's type must be assignable to its parameter's type.
1 TS2571: Object is of type 'unknown'.
1 TS2352: Conversion of type '{ capabilities
1 TS18049: 'errorInfo.componentStack' is possibly 'null' or 'undefined'.
```

I still see TS2352 and TS2604, but not TS2677

The descriptions are a bit different because I ran the previous one through chatgpt to get generalised descriptions rather than truncated ones

File: packages/js/product-editor/src/products.tsx [hunk 11]
```diff
@@ -1,6 +1,7 @@
 /**
  * External dependencies
  */
+import { Root } from 'react-dom/client';
```
Comment (ilyasfoo) [inline packages/js/product-editor/src/products.tsx:4]: Nit: more explicit 
```suggestion
import type { Root } from 'react-dom/client';
```

File: plugins/woocommerce-blocks/package.json [hunk 12]
```diff
@@ -307,7 +307,7 @@
 		"wordpress-components-slotfill": "npm:@wordpress/components@wp-6.5"
```
Comment (ilyasfoo) [inline plugins/woocommerce-blocks/package.json:310]: Should this be updated to 6.6 as well?

File: plugins/woocommerce-blocks/assets/js/base/context/hooks/collections/test/use-collection.jsx [hunk 13]
```diff
@@ -271,9 +271,7 @@ describe( 'useCollection', () => {
 		} );
 		const { results: results2 } = getProps( renderer );
 		expect( results2 ).toBe( results );
-		// expect 2 calls because internally, useSelect invokes callback twice
-		// on mount.
-		expect( mocks.selectors.getCollection ).toHaveBeenCalledTimes( 2 );
+		expect( mocks.selectors.getCollection ).toHaveBeenCalledTimes( 1 );
```
Comment (ilyasfoo) [inline plugins/woocommerce-blocks/assets/js/base/context/hooks/collections/test/use-collection.jsx:274]: Nice, I'm curious if this was fixed due to version update?

File: plugins/woocommerce-blocks/assets/js/blocks/cart-checkout-shared/payment-methods/test/payment-methods.js [hunk 14]
```diff
@@ -46,27 +46,30 @@ jest.mock( '@woocommerce/blocks-components', () => {
...
-		}
-		return originalStore;
-	} );
+jest.mock( '@wordpress/data', () => {
+	const originalModule = jest.requireActual( '@wordpress/data' );
+	return {
+		...originalModule,
+		select: jest.fn( ( storeName ) => {
+			const originalStore = originalModule.select( storeName );
+			if ( storeName === 'wc/store/payment' ) {
```
Comment (ilyasfoo) [inline plugins/woocommerce-blocks/assets/js/blocks/cart-checkout-shared/payment-methods/test/payment-methods.js:55]: I'm curious why this is changed from using constant `PAYMENT_STORE_KEY` to hardcoded string

File: plugins/woocommerce-blocks/assets/js/data/payment/test/selectors.js [hunk 15]
```diff
@@ -26,23 +26,28 @@ import {
...
-		};
-	}
-	return originalStore;
+jest.mock( '@wordpress/data', () => {
+	const originalModule = jest.requireActual( '@wordpress/data' );
+	return {
+		...originalModule,
+		select: jest.fn( ( storeName ) => {
+			const originalStore = originalModule.select( storeName );
+			if ( storeName === 'wc/store/cart' ) {
```
Comment (ilyasfoo) [inline plugins/woocommerce-blocks/assets/js/data/payment/test/selectors.js:35]: Same question as above

**Review by ilyasfoo — COMMENTED** [review]
Leaving another round of comments.

You probably already noticed this, but [CI has error](https://github.com/woocommerce/woocommerce/actions/runs/12387355281/job/34581710982?pr=53531) might be related to pnpm version update.

image

**Comment by rjchow** [comment]
> Leaving another round of comments.
> 
> You probably already noticed this, but [CI has error](https://github.com/woocommerce/woocommerce/actions/runs/12387355281/job/34581710982?pr=53531) might be related to pnpm version update.
> image

@ilyasfoo I asked about it here p1734512207475709-slack-C03CPM3UXDJ , it's because the workflow checks out a previous version of trunk to do a comparison. I don't think its within the scope of this PR to fix, though

File: plugins/woocommerce-blocks/assets/js/blocks/cart-checkout-shared/payment-methods/test/payment-methods.js (hunk 14 above)
Comment (rjchow) [reply plugins/woocommerce-blocks/assets/js/blocks/cart-checkout-shared/payment-methods/test/payment-methods.js:55]: jest.mock() doesn't allow variables from outside its scope https://github.com/jestjs/jest/issues/2567

File: plugins/woocommerce-blocks/assets/js/base/context/hooks/collections/test/use-collection.jsx (hunk 13 above)
Comment (rjchow) [reply plugins/woocommerce-blocks/assets/js/base/context/hooks/collections/test/use-collection.jsx:274]: Didn't check tbh but I think it's not a very useful assertion

File: plugins/woocommerce-blocks/package.json (hunk 12 above)
Comment (rjchow) [reply plugins/woocommerce-blocks/package.json:310]: It seemed like there's a specific version required originating from [this PR](https://github.com/woocommerce/woocommerce/pull/47105) so I didn't want to touch it - it would change the build

**Comment by Stojdza** [comment]
Hi @rjchow 👋 I'll go ahead and remove Solaris as the reviewer, as we were added by automation. But feel free to reassign to us if you think there's something specifically from the E2E and/or API tests you think we should look at in this PR. Thanks!

**Comment by ralucaStan** [comment]
> > Do think it's suboptimal and should be looked into, though.
> 
> this should be highlighted to the monorepo team

should be fixed with https://github.com/woocommerce/woocommerce/pull/53822

File: plugins/woocommerce/client/admin/webpack.config.js (hunk 10 above)
Comment (ralucaStan) [reply plugins/woocommerce/client/admin/webpack.config.js:209]: I wasn't running the command on all the repo areas, ~~  that's why I wasn't seeing those codes ~~ .

@rjchow you are running that command in `woocommerce/plugins/woocommerce-blocks`, right? I don't see any other TS errors in other plugins folders

**Comment by ralucaStan** [comment]
> Yesterday `pnpm-lock.yaml` had changes after I tested the branch.

I still see this file changing when switching to the branch

**Review by ralucaStan — APPROVED** [review]
I'm approving this PR. The blocks related tests are now passing. I've also smoke tested the checkout area.

Let me know if you need anything else to merge this PR

File: plugins/woocommerce/client/admin/webpack.config.js (hunk 10 above)
Comment (rjchow) [reply plugins/woocommerce/client/admin/webpack.config.js:209]: I was running the command in `./plugins/woocommerce/client/admin` - we're commenting on the webpack config that builds `admin` 😅 

It did not catch my attention that woocommerce-blocks might have TS errors since it didn't error when I was running build on the repo. Now that I'm aware, running `npx tsc` returns a large list of errors. I'm happy to take on the work of assigning/delegating for JS packages and `admin`, if you'd like to focus on `woocommerce-blocks`?

```
568 TS2339: Property 'getValidationErrorId' does not exist on type 'never'.
265 TS2739: Type '{ icon
225 TS7006: Parameter '_node' implicitly has an 'any' type.
225 TS2322: Type '{ [k
137 TS7005: Variable 'props' implicitly has an 'any' type.
137 TS2345: Argument of type '{ images
119 TS2769: No overload matches this call.
94 TS2554: Expected 2 arguments, but got 1.
88 TS7031: Binding element 'showItems' implicitly has an 'any' type.
75 TS2742: The inferred type of 'Default' cannot be named without a reference to '.pnpm/@storybook+types@7.5.2/node_modules/@storybook/types'. This is likely not portable. A type annotation is necessary.
64 TS2305: Module '"@wordpress/element"' has no exported member 'LazyExoticComponent'.
45 TS7016: Could not find a declaration file for module 'react-test-renderer'. '/Users/voidotter/Projects/automattic/woocommerce/woocommerce-monorepo/working/node_modules/.pnpm/react-test-renderer@18.3.1_react@18.3.1/node_modules/react-test-renderer/index.js' implicitly has an 'any' type.
36 TS7034: Variable 'renderer' implicitly has type 'any' in some locations where its type cannot be determined.
35 TS2741: Property 'slug' is missing in type '{ id
34 TS7053: Element implicitly has an 'any' type because expression of type '"Nonce-Timestamp"' can't be used to index type 'Object'.
32 TS18048: 'blocksConfig.productCount' is possibly 'undefined'.
31 TS18046: 'val' is of type 'unknown'.
23 TS2314: Generic type 'Array<T>' requires 1 type argument(s).
19 TS2724: '"@wordpress/block-editor"' has no exported member named 'useInnerBlocksProps'. Did you mean 'useBlockProps'?
16 TS2349: This expression is not callable.
15 TS2551: Property 'toHaveErrored' does not exist on type 'JestMatchers<Console>'. Did you mean 'toHaveRole'?
14 TS2578: Unused '@ts-expect-error' directive.
14 TS2307: Cannot find module '@wordpress/edit-site' or its corresponding type declarations.
12 TS2571: Object is of type 'unknown'.
12 TS2344: Type 'T' does not satisfy the constraint 'Record<string, any>'.
11 TS7015: Element implicitly has an 'any' type because index expression is not of type 'number'.
11 TS18047: 'width' is possibly 'null'.
10 TS2740: Type '{}' is missing the following properties from type 'ProductResponseImageItem'
10 TS2353: Object literal may only specify known properties, and ''data-foo'' does not exist in type 'HTMLProps<HTMLElement>'.
10 TS18049: 'args' is possibly 'null' or 'undefined'.
8 TS2375: Type '{ title
7 TS2379: Argument of type 'Element' is not assignable to parameter of type 'HTMLElement' with 'exactOptionalPropertyTypes
6 TS7019: Rest parameter 'rest' implicitly has an 'any[]' type.
6 TS2503: Cannot find namespace 'Gtag'.
5 TS2304: Cannot find name 'gtag'.
4 TS6307: File '/Users/voidotter/Projects/automattic/woocommerce/woocommerce-monorepo/working/plugins/woocommerce-blocks/packages/prices/utils/price.ts' is not listed within the file list of project '/Users/voidotter/Projects/automattic/woocommerce/woocommerce-monorepo/working/plugins/woocommerce-blocks/tsconfig.json'. Projects must list all files or use an 'include' pattern.
4 TS4082: Default export of the module has or is using private name 'WithReviewsState'.
4 TS2559: Type '{ products
4 TS2556: A spread argument must either have a tuple type or be passed to a rest parameter.
4 TS2540: Cannot assign to 'getProducts' because it is a read-only property.
3 TS7057: 'yield' expression implicitly results in an 'any' type because its containing generator lacks a return-type annotation.
3 TS2700: Rest types may only be created from object types.
3 TS2451: Cannot redeclare block-scoped variable 'store'.
2 TS7024: Function implicitly has return type 'any' because it does not have a return type annotation and is referenced directly or indirectly in one of its return expressions.
2 TS7023: 'getProductLayoutConfig' implicitly has return type 'any' because it does not have a return type annotation and is referenced directly or indirectly in one of its return expressions.
2 TS2786: 'OriginalComponent' cannot be used as a JSX component.
2 TS2698: Spread types may only be created from object types.
2 TS2694: Namespace '"/Users/voidotter/Projects/automattic/woocommerce/woocommerce-monorepo/working/node_modules/.pnpm/@wordpress+element@5.22.0/node_modules/@wordpress/element/build-types/index"' has no exported member 'ReactNode'.
2 TS2412: Type 'number[] | undefined' is not assignable to type 'string | string[] | undefined' with 'exactOptionalPropertyTypes
2 TS18043: Types cannot appear in export declarations in JavaScript files.
1 TS7022: 'children' implicitly has type 'any' because it does not have a type annotation and is referenced directly or indirectly in its own initializer.
1 TS5097: An import path can only end with a '.ts' extension when 'allowImportingTsExtensions' is enabled.
1 TS2749: 'Icon' refers to a value, but is being used as a type here. Did you mean 'typeof Icon'?
1 TS2561: Object literal may only specify known properties, but 'order' does not exist in type 'AttributeObject'. Did you mean to write 'orderby'?
1 TS2366: Function lacks ending return statement and return type does not include 'undefined'.
1 TS2351: This expression is not constructable.
1 TS2347: Untyped function calls may not accept type arguments.
1 TS1005: '}' expected.
```

Also, do we need to enable TS checking on `woocommerce-blocks` ? It doesn't seem to be part of the webpack config or `pnpm build`

**Comment by rjchow** [comment]
> > Yesterday `pnpm-lock.yaml` had changes after I tested the branch.
> 
> I still see this file changing when switching to the branch

Yup it's something that's happening on trunk against various versions of `pnpm` so I'll raise an issue for it - it's not something caused by this PR.

> I'm approving this PR. The blocks related tests are now passing. I've also smoke tested the checkout area.
> 
> Let me know if you need anything else to merge this PR
> 

I'll re-request reviews from ghidorah and once they've approved I'll merge it. Though I think we might want to address that TS issue [I mentioned above](https://github.com/woocommerce/woocommerce/pull/53531#discussion_r1891938486)

**Comment by rjchow** [comment]
(sorry @ralucaStan I didn't mean to re-request a review from you, there was some UI jankiness that resulted in a misclick)

**Review by chihsuan — APPROVED** [review]
Thanks, @rjchow, I've reviewed the latest changes and tested the flows again. LGTM!  🚀

File: plugins/woocommerce/client/admin/webpack.config.js (hunk 10 above)
Comment (ralucaStan) [reply plugins/woocommerce/client/admin/webpack.config.js:209]: @rjchow 
> I was running the command in ./plugins/woocommerce/client/admin - we're commenting on the webpack config that builds admin 😅

Double 😅😅 on my end, this was an obvious oversight on my part due to being stretched a bit thin

> I'm happy to take on the work of assigning/delegating for JS packages and admin, if you'd like to focus on woocommerce-blocks?

I can take care of the assigning/delegating. I'll share details in today's update

> Also, do we need to enable TS checking on woocommerce-blocks ? It doesn't seem to be part of the webpack config or pnpm build

Let's not do that now, I'll need to see when this was disabled, and I think we'd have to fix that outside of this project, as the blocks are already on React 18. I'll take care of documenting and delegation these as well.

File: packages/js/experimental/tsconfig-cjs.json [hunk 16]
```diff
@@ -2,6 +2,7 @@
 	"extends": "../tsconfig-cjs",
 	"compilerOptions": {
 		"outDir": "build",
+		"noCheck": true,
```
Comment (ralucaStan) [inline packages/js/experimental/tsconfig-cjs.json:5]: Why was this added @rjchow ?

Comment (ralucaStan) [reply packages/js/experimental/tsconfig-cjs.json:5]: I've answered myself. ` npx tsc` didn't show me errors, but I managed to see them in the end at build time.
